### PR TITLE
docs: describe message length varint

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/728

### CODE REVIEW

File: specs/src/specs/data_structures.md [hunk 1]
```diff
@@ -495,7 +495,8 @@ For shares **with a reserved namespace ID through [`NAMESPACE_ID_MAX_RESERVED`](
 For shares **with a namespace ID above [`NAMESPACE_ID_MAX_RESERVED`](./consensus.md#constants) but below [`PARITY_SHARE_NAMESPACE_ID`](./consensus.md#constants)**:
 
 - The first [`NAMESPACE_ID_BYTES`](./consensus.md#constants) of a share's raw data `rawData` is the namespace ID of that share, `namespaceID`.
-- The remaining [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_ID_BYTES`](./consensus.md#constants) bytes are request data. In other words, the remaining bytes have no special meaning and are simply used to store data.
+- If this is the first share of a message, the next 1-10 bytes contain a [varint](https://developers.google.com/protocol-buffers/docs/encoding) of the uint64 length of the message that follows. If this isn't the first share of a message, the next 1-10 bytes are used for request data.
```
Comment (SweeXordious) [inline specs/src/specs/data_structures.md:498]: [Non Blocking][Question]
I guess this is related to https://github.com/celestiaorg/celestia-app/pull/660/files, doesn't it make sense to link to it ? would be easier to know where all of this comes from

**Review by SweeXordious — COMMENTED** [review]
LGTM, only a question

File: specs/src/specs/data_structures.md (hunk 1 above)
Comment (rootulp) [reply specs/src/specs/data_structures.md:498]: the message length delimiter actually predates universal share prefix so I created separate issues:
1. https://github.com/celestiaorg/celestia-app/issues/728
2. https://github.com/celestiaorg/celestia-app/issues/729

In other words, this PR updates the spec to match the implementation prior to universal share prefix changes. A subsequent PR will update the spec to resolve #729

Comment (SweeXordious) [reply specs/src/specs/data_structures.md:498]: Aaaah, awesome, thanks a lot

**Review by SweeXordious — APPROVED** [review]
:tada:

**Comment by rootulp** [comment]
Will wait for @adlerjohn 's approval b/c he's the spec wizard

File: specs/src/specs/data_structures.md (hunk 1 above)
Comment (liamsi) [inline specs/src/specs/data_structures.md:498]: I think "request" is the wrong term here. @adlerjohn adapted it from the tendermint paper (which followed canonical BFT consensus research nomenclature). In our context, "request" is too vague or even confusing.

Comment (rootulp) [reply specs/src/specs/data_structures.md:498]: Agreed. Will fix in https://github.com/celestiaorg/celestia-app/pull/856#discussion_r993628147 after we align on better names

**Comment by rootulp** [comment]
Will merge this as-is to unblock https://github.com/celestiaorg/celestia-app/pull/856. If @adlerjohn has feedback, I can create follow-up PRs.

File: specs/src/specs/data_structures.md (hunk 1 above)
Comment (rootulp) [reply specs/src/specs/data_structures.md:498]: Following up here, fixed in https://github.com/celestiaorg/celestia-app/pull/856/commits/1d1ec47454f28526bdcfc44db95636f39e639fe7

**Review by egypjoy — APPROVED** [review]

**Pull request merged**